vioscsi/vioscsi.vcxproj

```diff
@@ -132,8 +132,30 @@
       <Command>
         copy /Y $(ProjectDir)vioscsi.DVL.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)
         copy /Y $(IntDir)vc.nativecodeanalysis.all.xml $(ProjectDir)
-        call "..\build\dvl1903.bat" "$(ProjectDir)\$(IntDir)" "$(TargetName)" "$(Configuration)" "$(Platform)"
-        copy /Y $(ProjectDir)vioscsi.DVL.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-win10.XML
+        if "$(TargetOS)"=="Win10" (
+          if exist C:\DVL1903 (
+            call "..\build\dvl1903.bat" "$(ProjectDir)\$(IntDir)" "$(TargetName)" "$(Configuration)" "$(Platform)"
+            copy /Y $(ProjectDir)vioscsi.DVL.XML $(ProjectDir)vioscsi.DVL-compat.XML 
+            copy /Y $(ProjectDir)vioscsi.DVL.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-compat.XML
+          )
+        )
+        if "$(TargetOS)"=="Win11" (
+          if exist $(ProjectDir)Install\Win10\$(TargetArch)\vioscsi.DVL.XML (
+            copy /Y $(ProjectDir)Install\Win10\$(TargetArch)\vioscsi.DVL.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-win10.XML
+          ) else (
+            copy /Y $(ProjectDir)vioscsi.DVL.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-win10.XML
+          )
+          if exist $(ProjectDir)Install\Win10\$(TargetArch)\vioscsi.DVL-compat.XML (
+            copy /Y $(ProjectDir)Install\Win10\$(TargetArch)\vioscsi.DVL-compat.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-compat.XML
+          ) else (
+            if exist $(ProjectDir)vioscsi.DVL-compat.XML (
+              copy /Y $(ProjectDir)vioscsi.DVL-compat.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-compat.XML
+            ) else (
+              findstr /v /c:"General.Checksum" "$(ProjectDir)vioscsi.DVL.XML" | findstr /v /c:".Semmle." > $(ProjectDir)vioscsi.DVL-compat.XML
+              copy /Y $(ProjectDir)vioscsi.DVL-compat.XML $(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-compat.XML
+            )
+          )
+        )
       </Command>
       <Inputs>$(IntDir)vc.nativecodeanalysis.all.xml;$(ProjectDir)vioscsi.DVL.XML</Inputs>
       <Outputs>$(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL.XML;$(ProjectDir)Install\$(TargetOS)\$(TargetArch)\vioscsi.DVL-win10.XML</Outputs>

```

Comment:
This is not a clean solution because not all <Outputs> files can be generated. Your code must generate everything that is specified in <Outputs> list but you skip some files or generate more conditionally.

Reply:
if "$(TargetOS)"=="Win10" and C:\DVL1903  does not exist `vioscsi.DVL-win10.XML` will be missing. 
Also script will generate `vioscsi.DVL-compat.XML` but it does not present in <Output>

Also `if exist C:\DVL1903 ` will mask the error that the folder does not exist.

Reply:
So when C:\DVL1903 exists, I end up with:

On Win10_SDV:
```
vioscsi.DVL.XML (Win10)
vioscsi.DVL-compat.XML (Win10 1903)
```
On Win11_SDV:
```
vioscsi.DVL.XML (Win11)
vioscsi.DVL-win10.XML (Win10)
vioscsi.DVL-compat.XML (Win10 1903)
```
So when C:\DVL1903 does NOT exist, I end up with:

On Win10_SDV:
```
vioscsi.DVL.XML (Win10)
vioscsi.DVL-compat.XML (fudged from Win10 and NOT Win10 1903)
```
On Win11_SDV:
```
vioscsi.DVL.XML (Win11)
vioscsi.DVL-win10.XML (Win10)
vioscsi.DVL-compat.XML (fudged from Win11 and NOT Win10 1903)
```
And without Win10_SDV, Win11_SDV produces:
```
vioscsi.DVL.XML (Win11)
vioscsi.DVL-win10.XML (Win11)
vioscsi.DVL-compat.XML (fudged from Win11 and NOT Win10 1903)
```
The above is with the `buildAll.bat` mods I'm proposing in #1212, which don't clobber any existing DVLs.

Presently, this is what happens when `C:\DVL1903` exists:

On Win10_SDV:
```
vioscsi.DVL.XML (Win10)
vioscsi.DVL-win10.XML (Win10 1903)
vioscsi.DVL-compat.XML (fudged from Win10 and NOT Win10 1903)
```
On Win11_SDV:
```
vioscsi.DVL.XML (Win11)
vioscsi.DVL-win10.XML (Win10 1903)
vioscsi.DVL-compat.XML (fudged from Win11 and NOT Win10 1903)
```
And when `C:\DVL1903` does NOT exist:

On Win10_SDV:
```
vioscsi.DVL.XML (Win10)
vioscsi.DVL-win10.XML (Win10)
vioscsi.DVL-compat.XML (fudged from Win10 and NOT Win10 1903)
```
On Win11_SDV:
```
vioscsi.DVL.XML (Win11)
vioscsi.DVL-win10.XML (Win11)
vioscsi.DVL-compat.XML (fudged from Win11 and NOT Win10 1903)
```
The end of `buildAll.bat` looks like this on output (`SKIP_SDV_ACTUAL=Yes`, so SDV variants aren't shown):

![virtio-win_dvl-compat](https://github.com/user-attachments/assets/f1de5ec4-ed28-4664-a043-ebfa35a23a81)

So I guess I need to check which one should be the `DVL1903` version: Is it `DVL-win10` or `DVL-compat`...?

Do you have a process that actually relies on the `The system cannot find the path specified.` error message in `:VCEnd` when `C:\DVL1903` does NOT exist..!?

Reply:
>Do you have a process that actually relies on the The system cannot find the path specified. error message in :VCEnd when C:\DVL1903 does NOT exist..!?

This is useful for debugging. After your changes, we always have all DVL (real or fake) and this is more complicated to understand why the Static tools test failed. 
With this log we can see that some file was missing during DVL generation.